modules/analysis/src/endpoints/test.rs

```diff
@@ -656,37 +656,27 @@ async fn spdx_package_of(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
     let response: Value = app.call_and_read_body_json(request).await;
     log::debug!("{response:#?}");
 
-    let sbom = &response["items"][0];
-    let matches: Vec<_> = sbom["deps"]
-        .as_array()
-        .into_iter()
-        .flatten()
-        .filter(|m| {
-            m == &&json!({
-                "sbom_id": sbom["sbom_id"],
-                "node_id": "SPDXRef-83c9faa0-ca85-4e48-9165-707b2f9a324b",
-                "relationship": "PackageOf",
-                "purl": [],
-                "cpe": m["cpe"], // long list assume it's correct
-                "name": "SATELLITE-6.15-RHEL-8",
-                "version": "6.15",
-                "deps": [],
-            })
-        })
-        .collect();
-
-    assert_eq!(1, matches.len());
+    assert_eq!(
+        response.pointer("/items/0/sbom_id"),
+        response.pointer("/items/0/deps/0/sbom_id")
```

Comment:
Don't hate it but I do think assuming the array position of the element could be considered brittle. Is that PackageOf relationship pointing at the expected package?

Reply:
You could include additional assertions about the lengths of the relevant arrays like I do in the test below, e.g.
```
assert!(response.pointer("/items/0/deps/1").is_none());
```
The `PackageOf` relationship is asserted on lines 663-666.

I don't like the pointer path as much as your `ContainsSubset` trait.